Packs/MongoDBAtlas/Integrations/MongoDBAtlas/MongoDBAtlas.py

```diff
@@ -0,0 +1,586 @@
+
+    Returns:
+        dict: A dictionary with `min_time` and `events_with_created_min_time` keys.
+    """
+    events_with_created_min_time = save_events_ids_with_specific_created_date(output, new_min_time)
+    return {'min_time': new_min_time,
+            'events_with_created_min_time': events_with_created_min_time
+            }
+
+
+def first_time_fetching_events(client: Client, fetch_limit: int):
+    """
+    Fetches and enriches the first batch of events, returning them with the minimum creation time.
+
+    Args:
+        client (Client): The client instance used to retrieve events.
+        fetch_limit (int): The maximum number of events to retrieve.
+
+    Returns:
+        tuple: A tuple containing:
+            - results (list): A list of event dictionaries.
+            - new_min_time (str): The creation time of the latest fetched event.
+    """
+    results = client.get_events_first_time_events(fetch_limit)
+    for event in results:
+        enrich_event(event, event_type='events')
+    last_fetched_event = results[0] if results else None
+    new_min_time = last_fetched_event.get('created') if last_fetched_event else None
+    return results, new_min_time
+
+
+def fetch_event_type(client: Client, fetch_limit: int, last_run: dict) -> tuple[list, dict]:
+    """
+    Fetches events until fetch_limit is reached, or no more events are available.
+
+    Args:
+        client (Client): MongoDBAtlas client.
+        fetch_limit (int): The maximum number of events to fetch.
+        last_run (dict): Dictionary containing data from the previous run.
+
+    Returns:
+        - The list containing all fetched events.
+        - The object to save for the next run.
+    """
+    min_date = last_run.get('min_time')
+    events_with_created_min_time = last_run.get('events_with_created_min_time') or []
+
+    demisto.debug(f'Start to fetch events with {min_date}')
+    if min_date:
+        results = client.get_events_request(min_date=min_date)
+    else:  # first time fetching events
+        results, new_min_time = first_time_fetching_events(client, fetch_limit)
+        new_last_run_obj = create_last_run_dict_for_events(results, new_min_time)
+        return results, new_last_run_obj
+
+    response = get_last_page_of_events(client, results)
+    links = response.get('links', [])
+    events = response.get('results', [])
+
+    current_fetched_events_amount = 0
+    output = []
+    new_min_date = min_date
+
+    while current_fetched_events_amount < fetch_limit:
+        for event in reversed(events):  # ignore
+            event_id = event.get('id')
+            if event_id in events_with_created_min_time:
+                continue
+
+            enrich_event(event, 'events')
+            output.append(event)
+            current_fetched_events_amount += 1
+            new_min_date = get_latest_date(new_min_date, event.get('created'))
+
+            demisto.debug(f'Fetched event ID {event_id} from type "events"')
```

Comment:
Let's remove this. We don't want 2500 lines of debug. 
Better just to add the len of events at the end.